aten/src/ATen/SparseTensorImpl.cpp

```diff
@@ -70,6 +70,7 @@ void SparseTensorImpl::set_storage_offset(int64_t storage_offset) {
 }
 
 int64_t SparseTensorImpl::dim() const {
+  TORCH_INTERNAL_ASSERT_DEBUG_ONLY(sparse_dim_ + dense_dim_ == TensorImpl::dim());
```

Comment:
why debug only?  Do we have full coverage for our debug builds?

Reply:
beats me, but slowing down `dim()` in prod seems like a bad choice.

Reply:
right, but you don't have to land it.  This would just ensure our OSS CI at least runs over it.

Reply:
done https://github.com/pytorch/pytorch/pull/50172